webhook_server/app.py

```diff
@@ -0,0 +1,42 @@
+from flask import Flask, request, Response, stream_with_context
+import os
+from queue import Queue
+import json
+import time
+
+app = Flask(__name__)
+app.secret_key = os.urandom(24)
+
+user_queues = {}
+
+@app.route('/webhook', methods=['GET'])
+def verify():
+    return request.args.get('challenge')
+
+@app.route('/webhook', methods=['POST'])
+def webhook():
+    # Verify the message is from Dropbox (how to store the APP_SECRET?)
+    # signature = request.headers.get('X-Dropbox-Signature')
+    # if not hmac.compare_digest(signature, hmac.new(APP_SECRET, request.data, sha256).hexdigest()):
+    #     abort(403)
+    
+    for userid in json.loads(request.data)['list_folder']['accounts']:
+        if userid not in user_queues:
+            user_queues[userid] = Queue()
+        user_queues[userid].put('File change detected')
+    return '', 200
+
+def stream_events(userid):
+    while True:
+        if userid in user_queues and not user_queues[userid].empty():
+            message = user_queues[userid].get()
+            yield f"data: {message}\n\n"
+        yield ":heartbeat\n\n"
+        time.sleep(20)
```

Comment:
Queueu.get() is blocking. No need time.sleep()

Reply:
This sleep is mainly for reduce the frequency of sending heartbeat. Could adjust the time of sleep according to the performance in later testing (likely to reduce the time of sleep)